File: leetcode75/345. Reverse Vowels of a String/solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        vowels = set('aeiouAEIOU')
        chars = list(s)  # Convert string to list of characters
        left, right = 0, len(s) - 1

        while left < right:
            while left < right and chars[left] not in vowels:
                left += 1
            
            while left < right and chars[right] not in vowels:
                right -= 1
            
            if left < right:
                chars[left], chars[right] = chars[right], chars[left]
                left += 1
                right -= 1

        return ''.join(chars)

File: leetcode75/345. Reverse Vowels of a String/Solution.cs
public class Solution {
    private static readonly HashSet<char> Vowels = new HashSet<char> { 'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U' };

    public string ReverseVowels(string s) {
        return string.Create(s.Length, s, (span, original) => { 
            int left = 0, right = s.Length - 1;

            while (left < right) {
                while (left < right && !Vowels.Contains(original[left])) {
                    span[left] = original[left];
                    left++;
                }

                while (left < right && !Vowels.Contains(original[right])) {
                    span[right] = original[right];
                    right--;
                }

                if (left < right) {
                    span[left] = original[right];
                    span[right] = original[left];
                    left++;
                    right--;
                }
            }

            // Copy any remaining characters
            while (left <= right) {
                span[left] = original[left];
                left++;
            }
        });
    }
}
